src/plugins/clips-executive/clips/resource-locks.clp

```diff
@@ -33,6 +33,18 @@
   (return FALSE)
 )
 
+(defrule resource-locks-append-action-resources
+  (declare (salience 1))
+  ?g <- (goal (mode COMMITTED)
```

Comment:
I think adding the resources when a goal is already committed is too late. In general, the decision to really commit to a goal once it is expanded might also be based on the required resources.